variables.tf

```
@@ -0,0 +1,38 @@
+variable "subscriptions" {
+  type        = map(string)
+  description = <<-DOC
+  ```
+  [
+    "<subscription_name>" = "<subscription_id>"    (list of subscriptions to recieve default group assignments) 
+  ]
+  ```
+  DOC
+  default     = {}
+}
+
+variable "management_groups" {
+  type = map(object({
+    display_name = string
+  }))
+  description = <<-DOC
+  ```
+  [
+    "<management_group_name>" = "<management_group_id>"    (list of management groups to recieve default group assignments) 
+  ]
+  ```
+  DOC
+  default     = {}
+}
+
+variable "custom_assignments" {
+  type        = map(map(list(string)))
```

oh wait, i see. You need to use a map, because roleidentifier is dynamic. And pim enabled is in turn dynamic, because it's a map of map of lists. I would like to propose a typed solution:

```suggestion
  type        = map(object({
  pim_enabled=optional(bool)
  management_groups = optional(map(string))
  subscriptions = optional(map(string))
  custom_scope = optional(map(string))
  }))
```